File: php/envio-com-anexo.php
<?php
// Inicializa a sessão cURL
$curl = curl_init();

// Define as opções cURL usando um array associativo
curl_setopt_array($curl, array(
    // URL da API Mailgrid para envio de email
    CURLOPT_URL => 'https://api.mailgrid.net.br/send/',
    
    // Faz com que o cURL retorne a resposta como string
    CURLOPT_RETURNTRANSFER => true,

    // Define a codificação de resposta para ser aceita
    CURLOPT_ENCODING => '',

    // Define o número máximo de redirecionamentos
    CURLOPT_MAXREDIRS => 10,

    // Define o tempo máximo de espera da requisição
    CURLOPT_TIMEOUT => 0,

    // Permite o cURL seguir redirecionamentos automaticamente
    CURLOPT_FOLLOWLOCATION => true,

    // Define a versão do protocolo HTTP a ser usado
    CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,

    // Define o método da requisição como POST
    CURLOPT_CUSTOMREQUEST => 'POST',

    // Define os dados do corpo da requisição em formato JSON
    CURLOPT_POSTFIELDS => '{
        "host_smtp": "HOST-SMTP", // Substitua por seu host SMTP
        "usuario_smtp": "USUARIO-SMTP", // Substitua pelo usuário SMTP
        "senha_smtp": "SENHA-SMTP", // Substitua pela senha SMTP
        "emailRemetente": "EMAIL-REMETENTE", // Email do remetente
        "nomeRemetente": "NOME-REMETENTE", // Nome do remetente
        "emailDestino": ["destinatario1@dominio.com.br","destinatario2@dominio.com.br"], // Emails dos destinatários
        "assunto": "Teste de envio com anexo via API JSON", // Assunto do email
        "mensagemAnexos": {
            "file1": {
                "name": "anexoexemplo.pdf", // Nome do arquivo
                "type": "application/pdf", // Tipo MIME do arquivo
                "content": "UEAEAAAAAAAAAAAAAAAAAAAAA/8QAFgEBAQEAAAAAAAAAAAAAAAAAAAYH/8QAFBEBAAAAAAAAAAAAAAAAAAAAAP/aAAwDAQACEQMRAD8AoytOgAAAAAAAP//Z" // Conteúdo do arquivo codificado em base64
            }
        },
        "mensagem": "<h1>Mensagem de teste da API com anexo</h1> <p>Atenção, o arquivo codificado acima não funciona, é apenas um exemplo.</p>", // Corpo da mensagem em HTML
        "mensagemTipo": "html", // Tipo de mensagem (HTML)
        "mensagemEncoding": "base64", // Codificação do arquivo anexo (base64)
        "mensagemAlt": "Mensagem de teste da API com anexo. >Atenção, o arquivo codificado acima não funciona, é apenas um exemplo." // Texto alternativo para clientes de email sem suporte a HTML
    }',

    // Define o cabeçalho da requisição, informando que o corpo está em JSON
    CURLOPT_HTTPHEADER => array(
        'Content-Type: application/json'
    ),
));

// Executa a requisição cURL e armazena a resposta
$response = curl_exec($curl);

// Fecha a sessão cURL
curl_close($curl);

// Exibe a resposta recebida da API
echo $response;
?>

File: C#/envio-com-anexo.cs

using System;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;

class Program

 // URL do endpoint da API
 private static readonly string apiUrl = "https://api.mailgrid.net.br/sendmail/";

  // Método principal assíncrono
	static async Task Main(string[] args)
	{
	// Cria a instância do HttpClient
	using (var client = new HttpClient())
	{
	// Cria a requisição HTTP com o método POST para a URL do endpoint
	var request = new HttpRequestMessage(HttpMethod.Post, apiUrl);

	// Dados do corpo da requisição, incluindo as informações do e-mail e anexos
	var jsonContent = "{\r\n" +
		"    \"host_smtp\": \"HOST-SMTP\",\r\n" +
		"    \"usuario_smtp\": \"USUARIO-SMTP\",\r\n" +
		"    \"senha_smtp\": \"SENHA-SMTP\",\r\n" +
		"    \"emailRemetente\": \"EMAIL-REMETENTE\",\r\n" +
		"    \"nomeRemetente\": \"NOME-REMETENTE\",\r\n" +
		"    \"emailDestino\": [\"dev@mailgrid.net.br\",\"postmaster@mailgrid.net.br\"],\r\n" +
		"    \"assunto\": \"Teste da API JSON com anexo\",\r\n" +
		"    \"mensagemAnexos\": [\r\n" +
		"       {\r\n" +
		"            \"name\": \"pixel.jpg\",\r\n" +
		"            \"type\": \"image/jpeg\",\r\n" +
		"            \"content\": \"/9j/4AAQSkZJRgABAQEBLAEsAAD/2wBDAAYEBAQFBAYFBQYJBgUGCQsIBgYIC...\"\r\n" +
		"       },\r\n" +
		"       {\r\n" +
		"            \"name\": \"pixel2.jpg\",\r\n" +
		"            \"type\": \"image/jpeg\",\r\n" +
		"            \"content\": \"/9j/4AAQSkZJRgABAQEBLAEsAAD/2wBDAAYEBAQFBAYFBQYJBgUGCQsIBgYIC...\"\r\n" +
		"    ],\r\n" +

		     "Mensagem de teste da API com anexos. Testando anexos de html no envio da api\",\r\n" +
		"    \"mensagem\": 
		"    \"mensagemTipo\": \"html\",\r\n" +
		"    \"mensagemEncoding\": \"quoted-printable\",\r\n" +
		"    \"mensagemAlt\": \"mensagem de teste da API JSON com anexos\"\r\n" +
		"    }";

	// Define o conteúdo da requisição como JSON
	var content = new StringContent(jsonContent, Encoding.UTF8, "application/json");

	// Adiciona o conteúdo ao corpo da requisição
	request.Content = content;

	try
	{
	 // Envia a requisição HTTP de forma assíncrona
	 var response = await client.SendAsync(request);

	 // Garante que a resposta tenha sido bem-sucedida (código de status 2xx)
	 response.EnsureSuccessStatusCode();

	 // Exibe o conteúdo da resposta da API
	 Console.WriteLine("Resposta da API:");
	 string responseContent = await response.Content.ReadAsStringAsync();
	 Console.WriteLine(responseContent);
	}
	catch (HttpRequestException ex)
	{
	 // Trata erros de requisição HTTP (como problemas de conexão ou falhas no servidor)
	 Console.WriteLine($"Erro ao fazer a requisição: {ex.Message}");
	}
	catch (Exception ex)
	{
		// Trata outros tipos de exceção
		Console.WriteLine($"Erro inesperado: {ex.Message}");
	 }
    }
  } 
}
							
